File: rx/receiver.js
Rx = require("rx.time");
Rx = require("rx.vertx");
var consumer = vertx.eventBus().consumer("heat-sensor").bodyStream();
var observable = Rx.Observable.fromReadStream(consumer);
observable.
  bufferWithTime(1000).
  filter(function (arr) { return arr.length > 0; }).
  map(function (arr) { return arr.reduce(function (acc, x) { return acc + x; }, 0) / arr.length; }).
  subscribe(function (heat) {
	  console.log('Current heat is: ' + heat);
});
console.log("listening");

File: rx/Receiver.java
import io.vertx.lang.rxjava.AbstractVerticle ;
import rx.Observable;
import java.util.concurrent.TimeUnit;
import java.util.stream.Collectors;

public class Receiver extends AbstractVerticle {

  public void start() {
    Observable<Double> observable = vertx.eventBus().<Double>consumer("heat-sensor").bodyStream().toObservable();
    observable.
         buffer(1, TimeUnit.SECONDS).
         map(samples -> samples.stream().
         collect(Collectors.averagingDouble(d -> d))).
         subscribe(heat -> {
      System.out.println("Current heat is " + heat);
      // vertx.eventBus().send("news-feed", "Current heat is " + heat);
    });
  }
}
